polars/polars-io/src/csv/write_impl.rs

```diff
@@ -129,7 +131,7 @@ fn write_anyvalue(
         #[cfg(feature = "dtype-datetime")]
         AnyValue::Datetime(_, _, tz) => {
             // If this is a datetime, then datetime_format was either set or inferred.
-            let datetime_format = datetime_format.unwrap_or_default();
+            let datetime_format = unsafe { *datetime_formats.get_unchecked(i) }.unwrap();
```

Comment:
This `unsafe` should be on the function.

If we `unwrap` the `Option<T>` in `datetime_formats`, why don't we fill it with `T` itself? That saves another branch in a hot loop.